<a href="https://colab.research.google.com/github/senapathi402-star/IDMB_CASE_STUDY/blob/main/IDMB_Case_Study.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **IDMB Case Study**

In [51]:
!gdown 1s2TkjSpzNc4SyxqRrQleZyDIHlc7bxnd
!gdown 1Ws-_s1fHZ9nHfGLVUQurbHDvStePlEJm

Downloading...
From: https://drive.google.com/uc?id=1s2TkjSpzNc4SyxqRrQleZyDIHlc7bxnd
To: /content/movies.csv
100% 112k/112k [00:00<00:00, 100MB/s]
Downloading...
From: https://drive.google.com/uc?id=1Ws-_s1fHZ9nHfGLVUQurbHDvStePlEJm
To: /content/directors.csv
100% 65.4k/65.4k [00:00<00:00, 89.2MB/s]


In [52]:
import pandas as pd
import numpy as np

In [53]:
movies = pd.read_csv('movies.csv',index_col = 0)
directors = pd.read_csv('directors.csv',index_col = 0)

In [54]:
movies

id     budget  popularity     revenue  \
0     43597  237000000         150  2787965087   
1     43598  300000000         139   961000000   
2     43599  245000000         107   880674609   
3     43600  250000000         112  1084939099   
5     43602  258000000         115   890871626   
...     ...        ...         ...         ...   
4736  48363          0           3      321952   
4743  48370      27000          19     3151130   
4748  48375          0           7           0   
4749  48376          0           3           0   
4768  48395     220000          14     2040920   

                                         title  vote_average  vote_count  \
0                                       Avatar           7.2       11800   
1     Pirates of the Caribbean: At World's End           6.9        4500   
2                                      Spectre           6.3        4466   
3                        The Dark Knight Rises           7.6        9106   
5                                 Spider-Man 3           5.9        3576   
...                                        ...           ...         ...   
4736                            The Last Waltz           7.9          64   
4743                                    Clerks           7.4         755   
4748                                   Rampage           6.0         131   
4749                                   Slacker           6.4          77   
4768                               El Mariachi           6.6         238   

      director_id  year month       day  
0            4762  2009   Dec  Thursday  
1            4763  2007   May  Saturday  
2            4764  2015   Oct    Monday  
3            4765  2012   Jul    Monday  
5            4767  2007   May   Tuesday  
...           ...   ...   ...       ...  
4736         4809  1978   May    Monday  
4743         5369  1994   Sep   Tuesday  
4748         5148  2009   Aug    Friday  
4749         5535  1990   Jul    Friday  
4768         5097  1992   Sep    Friday  

[1465 rows x 11 columns]

In [55]:
directors

director_name    id gender
0         James Cameron  4762   Male
1        Gore Verbinski  4763   Male
2            Sam Mendes  4764   Male
3     Christopher Nolan  4765   Male
4        Andrew Stanton  4766   Male
...                 ...   ...    ...
2344      Shane Carruth  7106   Male
2345   Neill Dela Llana  7107    NaN
2346        Scott Smith  7108    NaN
2347        Daniel Hsia  7109   Male
2348   Brian Herzlinger  7110   Male

[2349 rows x 3 columns]

In [56]:
df = movies.merge(directors,left_on = 'director_id',right_on = 'id',how = 'left')
df.head()

id_x     budget  popularity     revenue  \
0  43597  237000000         150  2787965087   
1  43598  300000000         139   961000000   
2  43599  245000000         107   880674609   
3  43600  250000000         112  1084939099   
4  43602  258000000         115   890871626   

                                      title  vote_average  vote_count  \
0                                    Avatar           7.2       11800   
1  Pirates of the Caribbean: At World's End           6.9        4500   
2                                   Spectre           6.3        4466   
3                     The Dark Knight Rises           7.6        9106   
4                              Spider-Man 3           5.9        3576   

   director_id  year month       day      director_name  id_y gender  
0         4762  2009   Dec  Thursday      James Cameron  4762   Male  
1         4763  2007   May  Saturday     Gore Verbinski  4763   Male  
2         4764  2015   Oct    Monday         Sam Mendes  4764   Male  
3         4765  2012   Jul    Monday  Christopher Nolan  4765   Male  
4         4767  2007   May   Tuesday          Sam Raimi  4767   Male

In [57]:
df.drop(['id_x','id_y','director_id'],axis = 1,inplace = True)

In [58]:
df.head()

budget  popularity     revenue  \
0  237000000         150  2787965087   
1  300000000         139   961000000   
2  245000000         107   880674609   
3  250000000         112  1084939099   
4  258000000         115   890871626   

                                      title  vote_average  vote_count  year  \
0                                    Avatar           7.2       11800  2009   
1  Pirates of the Caribbean: At World's End           6.9        4500  2007   
2                                   Spectre           6.3        4466  2015   
3                     The Dark Knight Rises           7.6        9106  2012   
4                              Spider-Man 3           5.9        3576  2007   

  month       day      director_name gender  
0   Dec  Thursday      James Cameron   Male  
1   May  Saturday     Gore Verbinski   Male  
2   Oct    Monday         Sam Mendes   Male  
3   Jul    Monday  Christopher Nolan   Male  
4   May   Tuesday          Sam Raimi   Male

In [59]:
df.shape

(1465, 11)

In [60]:
df.describe()

budget   popularity       revenue  vote_average    vote_count  \
count  1.465000e+03  1465.000000  1.465000e+03   1465.000000   1465.000000   
mean   4.802295e+07    30.855973  1.432539e+08      6.368191   1146.396587   
std    4.935541e+07    34.845214  2.064918e+08      0.818033   1578.077438   
min    0.000000e+00     0.000000  0.000000e+00      3.000000      1.000000   
25%    1.400000e+07    11.000000  1.738013e+07      5.900000    216.000000   
50%    3.300000e+07    23.000000  7.578164e+07      6.400000    571.000000   
75%    6.600000e+07    41.000000  1.792469e+08      6.900000   1387.000000   
max    3.800000e+08   724.000000  2.787965e+09      8.300000  13752.000000   

              year  
count  1465.000000  
mean   2002.615017  
std       8.680141  
min    1976.000000  
25%    1998.000000  
50%    2004.000000  
75%    2009.000000  
max    2016.000000

In [61]:
df.describe(include = object).T

count unique               top  freq
title          1465   1465       El Mariachi     1
month          1465     12               Dec   193
day            1465      7            Friday   654
director_name  1465    199  Steven Spielberg    26
gender         1341      2              Male  1309

In [62]:
df.head()

budget  popularity     revenue  \
0  237000000         150  2787965087   
1  300000000         139   961000000   
2  245000000         107   880674609   
3  250000000         112  1084939099   
4  258000000         115   890871626   

                                      title  vote_average  vote_count  year  \
0                                    Avatar           7.2       11800  2009   
1  Pirates of the Caribbean: At World's End           6.9        4500  2007   
2                                   Spectre           6.3        4466  2015   
3                     The Dark Knight Rises           7.6        9106  2012   
4                              Spider-Man 3           5.9        3576  2007   

  month       day      director_name gender  
0   Dec  Thursday      James Cameron   Male  
1   May  Saturday     Gore Verbinski   Male  
2   Oct    Monday         Sam Mendes   Male  
3   Jul    Monday  Christopher Nolan   Male  
4   May   Tuesday          Sam Raimi   Male

## change gender male as 1 and female as 0?

In [63]:
def encode(gender):
   if gender == 'Male':
      return 1
   else:
      return 0

In [64]:
df['gender'].apply(encode)

0       1
1       1
2       1
3       1
4       1
       ..
1460    1
1461    1
1462    1
1463    1
1464    0
Name: gender, Length: 1465, dtype: int64

In [65]:
df['gender'] = df['gender'].apply(encode)

In [66]:
df.head()

budget  popularity     revenue  \
0  237000000         150  2787965087   
1  300000000         139   961000000   
2  245000000         107   880674609   
3  250000000         112  1084939099   
4  258000000         115   890871626   

                                      title  vote_average  vote_count  year  \
0                                    Avatar           7.2       11800  2009   
1  Pirates of the Caribbean: At World's End           6.9        4500  2007   
2                                   Spectre           6.3        4466  2015   
3                     The Dark Knight Rises           7.6        9106  2012   
4                              Spider-Man 3           5.9        3576  2007   

  month       day      director_name  gender  
0   Dec  Thursday      James Cameron       1  
1   May  Saturday     Gore Verbinski       1  
2   Oct    Monday         Sam Mendes       1  
3   Jul    Monday  Christopher Nolan       1  
4   May   Tuesday          Sam Raimi       1

In [67]:
df['gender'].value_counts()

gender
1    1309
0     156
Name: count, dtype: int64

## Movies directed by each director?

In [68]:
df.groupby('director_name')['title'].count().head()

director_name
Adam McKay                     6
Adam Shankman                  8
Alejandro González Iñárritu    6
Alex Proyas                    5
Alexander Payne                5
Name: title, dtype: int64

In [69]:
# Find the director who have directed most number of movies
df.groupby('director_name')['title'].count().sort_values(ascending= False)[:1]

director_name
Steven Spielberg    26
Name: title, dtype: int64

In [70]:
df.groupby('director_name').groups

{'Adam McKay': [176, 323, 366, 505, 839, 916], 'Adam Shankman': [265, 300, 350, 404, 458, 843, 999, 1231], 'Alejandro González Iñárritu': [106, 749, 1015, 1034, 1077, 1405], 'Alex Proyas': [95, 159, 514, 671, 873], 'Alexander Payne': [793, 1006, 1101, 1211, 1281], 'Andrew Adamson': [11, 43, 328, 501, 947], 'Andrew Niccol': [533, 603, 701, 722, 1439], 'Andrzej Bartkowiak': [349, 549, 754, 911, 924], 'Andy Fickman': [517, 681, 909, 926, 973, 1023], 'Andy Tennant': [314, 320, 464, 593, 676, 885], 'Ang Lee': [99, 134, 748, 840, 1089, 1110, 1132, 1184], 'Anne Fletcher': [610, 650, 736, 789, 1206], 'Antoine Fuqua': [310, 338, 424, 467, 576, 808, 818, 1105], 'Atom Egoyan': [946, 1128, 1164, 1194, 1347, 1416], 'Barry Levinson': [313, 319, 471, 594, 878, 898, 1013, 1037, 1082, 1143, 1185, 1345, 1378], 'Barry Sonnenfeld': [13, 48, 90, 205, 591, 778, 783], 'Ben Stiller': [209, 212, 547, 562, 850], 'Bill Condon': [102, 307, 902, 1233, 1381], 'Bobby Farrelly': [352, 356, 481, 498, 624, 630, 654, 80

In [71]:
df.groupby('director_name').ngroups

199

In [72]:
df.groupby('director_name').get_group('Steven Spielberg').head()

budget  popularity    revenue  \
37   185000000          75  786636033   
105  140000000          44  183345589   
110  132000000          48  591739379   
114  130000000          89  371940071   
166  102000000          65  358372926   

                                                 title  vote_average  \
37   Indiana Jones and the Kingdom of the Crystal S...           5.7   
105                                            The BFG           6.0   
110                                  War of the Worlds           6.2   
114                           The Adventures of Tintin           6.7   
166                                    Minority Report           7.1   

     vote_count  year month        day     director_name  gender  
37         2495  2008   May  Wednesday  Steven Spielberg       1  
105        1000  2016   Jun  Wednesday  Steven Spielberg       1  
110        2322  2005   Jun    Tuesday  Steven Spielberg       1  
114        2061  2011   Oct    Tuesday  Steven Spielberg       1  
166        2608  2002   Jun   Thursday  Steven Spielberg       1

In [73]:
df.head()

budget  popularity     revenue  \
0  237000000         150  2787965087   
1  300000000         139   961000000   
2  245000000         107   880674609   
3  250000000         112  1084939099   
4  258000000         115   890871626   

                                      title  vote_average  vote_count  year  \
0                                    Avatar           7.2       11800  2009   
1  Pirates of the Caribbean: At World's End           6.9        4500  2007   
2                                   Spectre           6.3        4466  2015   
3                     The Dark Knight Rises           7.6        9106  2012   
4                              Spider-Man 3           5.9        3576  2007   

  month       day      director_name  gender  
0   Dec  Thursday      James Cameron       1  
1   May  Saturday     Gore Verbinski       1  
2   Oct    Monday         Sam Mendes       1  
3   Jul    Monday  Christopher Nolan       1  
4   May   Tuesday          Sam Raimi       1

## Which director is most productive?

In [74]:
data = df.groupby('director_name').agg({'title':'count','year':['max','min']})

In [75]:
data

title  year      
                            count   max   min
director_name                                
Adam McKay                      6  2015  2004
Adam Shankman                   8  2012  2001
Alejandro González Iñárritu     6  2015  2000
Alex Proyas                     5  2016  1994
Alexander Payne                 5  2013  1999
...                           ...   ...   ...
Wes Craven                     10  2011  1984
Wolfgang Petersen               7  2006  1981
Woody Allen                    18  2013  1977
Zack Snyder                     7  2016  2004
Zhang Yimou                     6  2014  2002

[199 rows x 3 columns]

In [76]:
data['year_active'] = data['year','max']-data['year','min']

In [77]:
data

title  year       year_active
                            count   max   min            
director_name                                            
Adam McKay                      6  2015  2004          11
Adam Shankman                   8  2012  2001          11
Alejandro González Iñárritu     6  2015  2000          15
Alex Proyas                     5  2016  1994          22
Alexander Payne                 5  2013  1999          14
...                           ...   ...   ...         ...
Wes Craven                     10  2011  1984          27
Wolfgang Petersen               7  2006  1981          25
Woody Allen                    18  2013  1977          36
Zack Snyder                     7  2016  2004          12
Zhang Yimou                     6  2014  2002          12

[199 rows x 4 columns]

In [78]:
data['productivity'] = data['title','count']/data['year_active']

In [79]:
data.head()

title  year       year_active productivity
                            count   max   min                         
director_name                                                         
Adam McKay                      6  2015  2004          11     0.545455
Adam Shankman                   8  2012  2001          11     0.727273
Alejandro González Iñárritu     6  2015  2000          15     0.400000
Alex Proyas                     5  2016  1994          22     0.227273
Alexander Payne                 5  2013  1999          14     0.357143

In [80]:
final_data = data.reset_index(0)

In [81]:
final_data.sort_values(by ='productivity',ascending = False)[:1][['director_name','productivity']]

director_name productivity
                              
190   Tyler Perry     1.285714

## How many years of data is given to you in the movies dataset?

In [82]:
movies.head()

id     budget  popularity     revenue  \
0  43597  237000000         150  2787965087   
1  43598  300000000         139   961000000   
2  43599  245000000         107   880674609   
3  43600  250000000         112  1084939099   
5  43602  258000000         115   890871626   

                                      title  vote_average  vote_count  \
0                                    Avatar           7.2       11800   
1  Pirates of the Caribbean: At World's End           6.9        4500   
2                                   Spectre           6.3        4466   
3                     The Dark Knight Rises           7.6        9106   
5                              Spider-Man 3           5.9        3576   

   director_id  year month       day  
0         4762  2009   Dec  Thursday  
1         4763  2007   May  Saturday  
2         4764  2015   Oct    Monday  
3         4765  2012   Jul    Monday  
5         4767  2007   May   Tuesday

In [83]:
movies['year'].nunique()

41

**OR**

In [84]:
movies['year'].max()-movies['year'].min()

40

## Display the release date in the day-month-year format?

In [85]:
movies.info()

<class 'pandas.core.frame.DataFrame'>
Index: 1465 entries, 0 to 4768
Data columns (total 11 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   id            1465 non-null   int64  
 1   budget        1465 non-null   int64  
 2   popularity    1465 non-null   int64  
 3   revenue       1465 non-null   int64  
 4   title         1465 non-null   object 
 5   vote_average  1465 non-null   float64
 6   vote_count    1465 non-null   int64  
 7   director_id   1465 non-null   int64  
 8   year          1465 non-null   int64  
 9   month         1465 non-null   object 
 10  day           1465 non-null   object 
dtypes: float64(1), int64(7), object(3)
memory usage: 137.3+ KB


In [86]:
movies['year'] = movies['year'].astype(str)

In [87]:
movies.info()

<class 'pandas.core.frame.DataFrame'>
Index: 1465 entries, 0 to 4768
Data columns (total 11 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   id            1465 non-null   int64  
 1   budget        1465 non-null   int64  
 2   popularity    1465 non-null   int64  
 3   revenue       1465 non-null   int64  
 4   title         1465 non-null   object 
 5   vote_average  1465 non-null   float64
 6   vote_count    1465 non-null   int64  
 7   director_id   1465 non-null   int64  
 8   year          1465 non-null   object 
 9   month         1465 non-null   object 
 10  day           1465 non-null   object 
dtypes: float64(1), int64(6), object(4)
memory usage: 137.3+ KB


In [88]:
movies['release_date'] = movies['day']+'-'+movies['month']+'-'+movies['year'].astype(str)

In [89]:
movies.head()

id     budget  popularity     revenue  \
0  43597  237000000         150  2787965087   
1  43598  300000000         139   961000000   
2  43599  245000000         107   880674609   
3  43600  250000000         112  1084939099   
5  43602  258000000         115   890871626   

                                      title  vote_average  vote_count  \
0                                    Avatar           7.2       11800   
1  Pirates of the Caribbean: At World's End           6.9        4500   
2                                   Spectre           6.3        4466   
3                     The Dark Knight Rises           7.6        9106   
5                              Spider-Man 3           5.9        3576   

   director_id  year month       day       release_date  
0         4762  2009   Dec  Thursday  Thursday-Dec-2009  
1         4763  2007   May  Saturday  Saturday-May-2007  
2         4764  2015   Oct    Monday    Monday-Oct-2015  
3         4765  2012   Jul    Monday    Monday-Jul-2012  
5         4767  2007   May   Tuesday   Tuesday-May-2007

## What is the percentage of the male directors in the given data set?

In [90]:
directors.head()

director_name    id gender
0      James Cameron  4762   Male
1     Gore Verbinski  4763   Male
2         Sam Mendes  4764   Male
3  Christopher Nolan  4765   Male
4     Andrew Stanton  4766   Male

In [91]:
(directors['gender'].value_counts()/directors['gender'].count()*100).round(1)

gender
Male      91.3
Female     8.7
Name: count, dtype: float64

## How many different movies have been directed in the year 2010?

In [92]:
movies.head()

id     budget  popularity     revenue  \
0  43597  237000000         150  2787965087   
1  43598  300000000         139   961000000   
2  43599  245000000         107   880674609   
3  43600  250000000         112  1084939099   
5  43602  258000000         115   890871626   

                                      title  vote_average  vote_count  \
0                                    Avatar           7.2       11800   
1  Pirates of the Caribbean: At World's End           6.9        4500   
2                                   Spectre           6.3        4466   
3                     The Dark Knight Rises           7.6        9106   
5                              Spider-Man 3           5.9        3576   

   director_id  year month       day       release_date  
0         4762  2009   Dec  Thursday  Thursday-Dec-2009  
1         4763  2007   May  Saturday  Saturday-May-2007  
2         4764  2015   Oct    Monday    Monday-Oct-2015  
3         4765  2012   Jul    Monday    Monday-Jul-2012  
5         4767  2007   May   Tuesday   Tuesday-May-2007

In [93]:
movies[movies['year'] == '2010'].value_counts().sum()

np.int64(63)

**OR**

In [94]:
len(movies[movies['year'] == '2010'])

63

## How many different directors have directed a movie since 2010(included)?

In [95]:
df.head()

budget  popularity     revenue  \
0  237000000         150  2787965087   
1  300000000         139   961000000   
2  245000000         107   880674609   
3  250000000         112  1084939099   
4  258000000         115   890871626   

                                      title  vote_average  vote_count  year  \
0                                    Avatar           7.2       11800  2009   
1  Pirates of the Caribbean: At World's End           6.9        4500  2007   
2                                   Spectre           6.3        4466  2015   
3                     The Dark Knight Rises           7.6        9106  2012   
4                              Spider-Man 3           5.9        3576  2007   

  month       day      director_name  gender  
0   Dec  Thursday      James Cameron       1  
1   May  Saturday     Gore Verbinski       1  
2   Oct    Monday         Sam Mendes       1  
3   Jul    Monday  Christopher Nolan       1  
4   May   Tuesday          Sam Raimi       1

In [96]:
df1 = df[df['year']>=2010]
df1['director_name'].nunique()

163

## What are the names of those directors?

In [97]:
df1['director_name'].unique()

array(['Sam Mendes', 'Christopher Nolan', 'Zack Snyder', 'Gore Verbinski',
       'Rob Marshall', 'Barry Sonnenfeld', 'Peter Jackson',
       'Ridley Scott', 'Peter Berg', 'Tim Burton', 'Michael Bay',
       'Sam Raimi', 'Martin Campbell', 'James Wan', 'Marc Forster',
       'Bryan Singer', 'Mike Newell', 'Guillermo del Toro', 'Justin Lin',
       'Lilly Wachowski', 'Martin Scorsese', 'David Ayer', 'Doug Liman',
       'Jon Favreau', 'Roland Emmerich', 'Francis Lawrence',
       'Jon Turteltaub', 'Michael Apted', 'George Miller',
       'Kenneth Branagh', 'Jonathan Liebesman', 'M. Night Shyamalan',
       'Joe Wright', 'Rob Minkoff', 'Alex Proyas', 'Jon M. Chu',
       'Bill Condon', 'Steven Spielberg', 'Alejandro González Iñárritu',
       'Phillip Noyce', 'Darren Aronofsky', 'Shawn Levy',
       'Robert Schwentke', 'Guy Ritchie', 'Louis Leterrier', 'Ang Lee',
       'Paul Greengrass', 'James Mangold', 'Raja Gosnell', 'Paul Weitz',
       'Adam McKay', 'Quentin Tarantino', 'Simon West

## Display the name of the director who directed the "spider-man 3" movie?

In [98]:
df[df['title'] == 'Spider-Man 3']['director_name']

4    Sam Raimi
Name: director_name, dtype: object

## How many different movies has been directed by the "James Cameron" between 2005 to 2010?

In [99]:
df2 = df[df['year'].between(2005,2010)]
df2.head()

budget  popularity     revenue  \
0  237000000         150  2787965087   
1  300000000         139   961000000   
4  258000000         115   890871626   
6  270000000          57   391081192   
7  200000000         107   586090727   

                                      title  vote_average  vote_count  year  \
0                                    Avatar           7.2       11800  2009   
1  Pirates of the Caribbean: At World's End           6.9        4500  2007   
4                              Spider-Man 3           5.9        3576  2007   
6                          Superman Returns           5.4        1400  2006   
7                         Quantum of Solace           6.1        2965  2008   

  month        day   director_name  gender  
0   Dec   Thursday   James Cameron       1  
1   May   Saturday  Gore Verbinski       1  
4   May    Tuesday       Sam Raimi       1  
6   Jun  Wednesday    Bryan Singer       1  
7   Oct   Thursday    Marc Forster       1

In [100]:
df2[df2['director_name']== 'James Cameron']['title']

0    Avatar
Name: title, dtype: object

## Among the top 10 most popular movies which is the movie with 2nd highest vote_average?

In [101]:
top_10_popular = df.sort_values(by = 'popularity',ascending = False)[:10]
top_10_popular

budget  popularity     revenue  \
58   165000000         724   675120017   
78   150000000         434   378858340   
119  140000000         271   655011224   
120  125000000         206   752100229   
45   185000000         187  1004558444   
59   160000000         167   825532764   
162  108000000         167   630161890   
5    250000000         155   873260194   
0    237000000         150  2787965087   
383   63000000         146   100853753   

                                                 title  vote_average  \
58                                        Interstellar           8.1   
78                                  Mad Max: Fury Road           7.2   
119  Pirates of the Caribbean: The Curse of the Bla...           7.5   
120              The Hunger Games: Mockingjay - Part 1           6.6   
45                                     The Dark Knight           8.2   
59                                           Inception           8.1   
162                                        The Martian           7.6   
5                   Batman v Superman: Dawn of Justice           5.7   
0                                               Avatar           7.2   
383                                         Fight Club           8.3   

     vote_count  year month        day      director_name  gender  
58        10867  2014   Nov  Wednesday  Christopher Nolan       1  
78         9427  2015   May  Wednesday      George Miller       1  
119        6985  2003   Jul  Wednesday     Gore Verbinski       1  
120        5584  2014   Nov    Tuesday   Francis Lawrence       1  
45        12002  2008   Jul  Wednesday  Christopher Nolan       1  
59        13752  2010   Jul  Wednesday  Christopher Nolan       1  
162        7268  2015   Sep  Wednesday       Ridley Scott       1  
5          7004  2016   Mar  Wednesday        Zack Snyder       1  
0         11800  2009   Dec   Thursday      James Cameron       1  
383        9413  1999   Oct     Friday      David Fincher       1

In [102]:
top_10_popular.sort_values(by = 'vote_average',ascending = False)[1:2]

budget  popularity     revenue            title  vote_average  \
45  185000000         187  1004558444  The Dark Knight           8.2   

    vote_count  year month        day      director_name  gender  
45       12002  2008   Jul  Wednesday  Christopher Nolan       1

## How many movies that are among top 10 popular movies are also among top 10 rated "Vote_avearge"?

In [103]:
top_10_popular

budget  popularity     revenue  \
58   165000000         724   675120017   
78   150000000         434   378858340   
119  140000000         271   655011224   
120  125000000         206   752100229   
45   185000000         187  1004558444   
59   160000000         167   825532764   
162  108000000         167   630161890   
5    250000000         155   873260194   
0    237000000         150  2787965087   
383   63000000         146   100853753   

                                                 title  vote_average  \
58                                        Interstellar           8.1   
78                                  Mad Max: Fury Road           7.2   
119  Pirates of the Caribbean: The Curse of the Bla...           7.5   
120              The Hunger Games: Mockingjay - Part 1           6.6   
45                                     The Dark Knight           8.2   
59                                           Inception           8.1   
162                                        The Martian           7.6   
5                   Batman v Superman: Dawn of Justice           5.7   
0                                               Avatar           7.2   
383                                         Fight Club           8.3   

     vote_count  year month        day      director_name  gender  
58        10867  2014   Nov  Wednesday  Christopher Nolan       1  
78         9427  2015   May  Wednesday      George Miller       1  
119        6985  2003   Jul  Wednesday     Gore Verbinski       1  
120        5584  2014   Nov    Tuesday   Francis Lawrence       1  
45        12002  2008   Jul  Wednesday  Christopher Nolan       1  
59        13752  2010   Jul  Wednesday  Christopher Nolan       1  
162        7268  2015   Sep  Wednesday       Ridley Scott       1  
5          7004  2016   Mar  Wednesday        Zack Snyder       1  
0         11800  2009   Dec   Thursday      James Cameron       1  
383        9413  1999   Oct     Friday      David Fincher       1

In [104]:
top_10_rated = df.sort_values(by = 'vote_average',ascending = False)[:10]
top_10_rated

budget  popularity     revenue               title  vote_average  \
1284    8000000         121   213928762        Pulp Fiction           8.3   
383    63000000         146   100853753          Fight Club           8.3   
901    22000000         104   321365567    Schindler's List           8.3   
459    55000000         138   677945399        Forrest Gump           8.2   
45    185000000         187  1004558444     The Dark Knight           8.2   
917    25000000          63    46836394          GoodFellas           8.2   
785    33000000          79   327311859               Se7en           8.1   
1326    6000000          64    23341568  The Usual Suspects           8.1   
58    165000000         724   675120017        Interstellar           8.1   
59    160000000         167   825532764           Inception           8.1   

      vote_count  year month        day      director_name  gender  
1284        8428  1994   Oct   Saturday  Quentin Tarantino       1  
383         9413  1999   Oct     Friday      David Fincher       1  
901         4329  1993   Nov     Monday   Steven Spielberg       1  
459         7927  1994   Jul  Wednesday    Robert Zemeckis       1  
45         12002  2008   Jul  Wednesday  Christopher Nolan       1  
917         3128  1990   Sep  Wednesday    Martin Scorsese       1  
785         5765  1995   Sep     Friday      David Fincher       1  
1326        3254  1995   Jul  Wednesday       Bryan Singer       1  
58         10867  2014   Nov  Wednesday  Christopher Nolan       1  
59         13752  2010   Jul  Wednesday  Christopher Nolan       1

In [105]:
result = pd.DataFrame(set(top_10_popular['title']).intersection(set(top_10_rated['title'])))
result

0
0        Inception
1       Fight Club
2     Interstellar
3  The Dark Knight

## Create a new matrix called revenue_per_budget?

In [106]:
df['revenue_per_budget'] = (df['revenue']/df['budget']).round(2)
df

budget  popularity     revenue  \
0     237000000         150  2787965087   
1     300000000         139   961000000   
2     245000000         107   880674609   
3     250000000         112  1084939099   
4     258000000         115   890871626   
...         ...         ...         ...   
1460          0           3      321952   
1461      27000          19     3151130   
1462          0           7           0   
1463          0           3           0   
1464     220000          14     2040920   

                                         title  vote_average  vote_count  \
0                                       Avatar           7.2       11800   
1     Pirates of the Caribbean: At World's End           6.9        4500   
2                                      Spectre           6.3        4466   
3                        The Dark Knight Rises           7.6        9106   
4                                 Spider-Man 3           5.9        3576   
...                                        ...           ...         ...   
1460                            The Last Waltz           7.9          64   
1461                                    Clerks           7.4         755   
1462                                   Rampage           6.0         131   
1463                                   Slacker           6.4          77   
1464                               El Mariachi           6.6         238   

      year month       day      director_name  gender  revenue_per_budget  
0     2009   Dec  Thursday      James Cameron       1               11.76  
1     2007   May  Saturday     Gore Verbinski       1                3.20  
2     2015   Oct    Monday         Sam Mendes       1                3.59  
3     2012   Jul    Monday  Christopher Nolan       1                4.34  
4     2007   May   Tuesday          Sam Raimi       1                3.45  
...    ...   ...       ...                ...     ...                 ...  
1460  1978   May    Monday    Martin Scorsese       1                 inf  
1461  1994   Sep   Tuesday        Kevin Smith       1              116.71  
1462  2009   Aug    Friday           Uwe Boll       1                 NaN  
1463  1990   Jul    Friday  Richard Linklater       1                 NaN  
1464  1992   Sep    Friday   Robert Rodriguez       0                9.28  

[1465 rows x 12 columns]

In [107]:
df['revenue_per_budget'].head()

0    11.76
1     3.20
2     3.59
3     4.34
4     3.45
Name: revenue_per_budget, dtype: float64

## In which year did christoper nolan produced most number of movies?

In [108]:
df[df['director_name'] == 'Christopher Nolan']['year'].value_counts().reset_index().sort_values(by = 'year',ascending = True)

year  count
7  2000      1
5  2002      1
4  2005      1
6  2006      1
1  2008      1
3  2010      1
0  2012      1
2  2014      1

## In which decade, the most number of movies were produced?

In [109]:
((df['year']//10)*10).value_counts().sort_values(ascending=False)[:1]

year
2000    640
Name: count, dtype: int64

## Within the decade with number of movies, who were the top 5 contribution directors (who produced more)?


In [110]:
df['director_name'].value_counts().head()

director_name
Steven Spielberg    26
Martin Scorsese     19
Clint Eastwood      19
Woody Allen         18
Spike Lee           16
Name: count, dtype: int64

## Display movie details of only those who where number of movies produced are more than 70?

In [111]:
df.groupby("year").filter(lambda x: x['director_name'].count() > 70)

budget  popularity     revenue  \
6     270000000          57   391081192   
8     200000000         145  1065659812   
18    207000000          61   550000000   
24    210000000           3   459359555   
43    180000000          67   748806957   
...         ...         ...         ...   
1342    5000000          18    26888376   
1359          0           3           0   
1365    4200000          16           0   
1373    4000000          17    31584901   
1407    2000000          12     5967038   

                                                  title  vote_average  \
6                                      Superman Returns           5.4   
8            Pirates of the Caribbean: Dead Man's Chest           7.0   
18                                            King Kong           6.6   
24                                X-Men: The Last Stand           6.3   
43    The Chronicles of Narnia: The Lion, the Witch ...           6.7   
...                                                 ...           ...   
1342                                          Clerks II           6.9   
1359                              A Cock and Bull Story           6.2   
1365                                High School Musical           6.1   
1373                                              Scoop           6.4   
1407                                        Harsh Times           6.3   

      vote_count  year month        day         director_name  gender  \
6           1400  2006   Jun  Wednesday          Bryan Singer       1   
8           5246  2006   Jun    Tuesday        Gore Verbinski       1   
18          2337  2005   Dec  Wednesday         Peter Jackson       1   
24          3525  2006   May  Wednesday          Brett Ratner       1   
43          2629  2005   Dec  Wednesday        Andrew Adamson       1   
...          ...   ...   ...        ...                   ...     ...   
1342         398  2006   May   Thursday           Kevin Smith       1   
1359          43  2005   Jul     Sunday  Michael Winterbottom       1   
1365        1000  2006   Jan     Friday          Kenny Ortega       1   
1373         420  2006   Jul   Thursday           Woody Allen       1   
1407         197  2005   Sep     Sunday            David Ayer       0   

      revenue_per_budget  
6                   1.45  
8                   5.33  
18                  2.66  
24                  2.19  
43                  4.16  
...                  ...  
1342                5.38  
1359                 NaN  
1365                0.00  
1373                7.90  
1407                2.98  

[144 rows x 12 columns]

##**Insights**

- Total of 199 directors were present.
- "Steven Spielberg" is the one who directed most of the movies till now.
- "Tyler Perry" is the top 1 productive director.
- Total 40 year of data was produced.
- Male directors was in this data set was total of 91%.
- 63 movies were directed in the year of 2010.
- 163 movies were directed by different directors from 2010 to 2016.
- "James Cameron" directed 'Avatar' movie from 2005 to 2010.
- "The Dark Knight" was top 2nd rated movie among the top 10 movies in the whole dataset.
- 'Fight Club', 'Inception', 'Interstellar', 'The Dark Knight' these were the top 10 popluar and as well as top 10 rated movies.
- In the decade of 2000 most movies were released.
- In the decade of 2000 these were the top 5 directors who directed most movies 'Steven Spielberg','Martin Scorsese','Clint Eastwood','Woody Allen','Spike Lee'.
